text/2021-09-27-refine-error-handling.md

```diff
@@ -0,0 +1,293 @@
+# Refine Error Handling
+
+## Background
+
+There are a lot of different error handling pattern in Chaos Mesh. Some of the
+errors in the code are equipped with a backtrace, while some are not. The mass
+in error handling has been blocking us from diagnosing the problem and handling
+potential errors. The Chaos Mesh code base imports `github.com/pingcap/errors`,
+`github.com/pkg/errors` and the `errors` in standard library.
+
+## Proposal
+
+Most of the error handling pattern follows [uber
+go-style](https://github.com/uber-go/guide/blob/master/style.md), and some rules
+are modified or removed because of the latest update of `pkg/errors` and
+`errors` in standard library.
+
+### Error Types
+
+When returning errors, consider the following to determine the best choice:
+
+1. Does the client need to extract the information of error? If so, you should
+   use a custom type, and implement `Error()` method. You should also wrap it
```

Comment:
We may need an example about how to write a custom type of error.

Reply:
Example moved here 😄